vizro-ai/src/vizro_ai/dashboard/utils.py

```diff
@@ -61,3 +63,30 @@ def _register_data(all_df_metadata: AllDfMetadata) -> vm.Dashboard:
 
     for name, metadata in all_df_metadata.all_df_metadata.items():
         data_manager[name] = metadata.df
+
+
+def _dashboard_code(
+    dashboard: vm.Dashboard, extra_callable_defs: Optional[Set[str]] = None, extra_imports: Optional[Set[str]] = None
+) -> str:
+    """Get the code for the dashboard."""
+    return dashboard._to_python(extra_callable_defs=extra_callable_defs, extra_imports=extra_imports)
+
+
+def _process_to_set(list_of_list_of_dict):
+    custom_func = set()
+    imports = set()
+    for list_of_dict in list_of_list_of_dict:
+        for d in list_of_dict:
+            for value in d.values():
+                # Extract imports
+                import_match = re.match(r"((?:from|import).*?)\n\n", value, re.DOTALL)
```

Comment:
Is this reliable? i.e. are there always `\n\n` separating imports from the rest of the code?

Reply:
So far this is quite consistent. So when llm generate the code, it always has an empty line between imports and the "@capture('graph')".  So in total there are 2 "\n". We should be good at least for OpenAI models, but I don't know whether other llm has the same pattern.
Consider this function will be gone after Max's refactor, let me leave it as is now. But I will try with all current models and check the pattern